nodejs-7_x: 7.9.0 -> 7.10.0

lo1tuma opened the pull request:
###### Motivation for this change

package update

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

nodejs updated gyp in https://github.com/nodejs/node/pull/12450 so I had to update the `no-xcode.patch`. It would be nice if a few more macOS user could test this update and confirm that the patch is working.

lo1tuma committed:
nodejs-7_x: 7.9.0 -> 7.10.0

fstoerkle commented:
I tested this and could successfully build Node `v7.10.0` on macOS Sierra (`v10.12.4`, XCode was already installed)

lo1tuma commented:
I’ve just noticed that we use the same `no-xcode.patch` file for v4, v6 and v7 but gyp was only updated in v7 so the updated patch is also only required for v7.
I updated this PR and introduced a separate `no-xcode-v7.patch` file which will be only used for nodejs v7.
I’ve tested all 3 nodejs versions on darwin.

offlinehacker commented:
Tnx!
